akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -37,13 +36,60 @@ trait RetrySupport {
   def retry[T](attempt: () => Future[T], attempts: Int, delay: FiniteDuration)(
       implicit ec: ExecutionContext,
       scheduler: Scheduler): Future[T] = {
+    retry(attempt, attempts, _ => Some(delay))
+  }
+
+  /**
+   * Given a function from Unit to Future, returns an internally retrying Future.
+   * The first attempt will be made immediately, each subsequent attempt will be made after 'delay'.
+   * A scheduler (eg context.system.scheduler) must be provided to delay each retry.
+   * You could provide a function to generate the next delay duration after first attempt,
+   * this function should never return `null`, otherwise an [[IllegalArgumentException]] will be through.
+   * If attempts are exhausted the returned future is simply the result of invoking attempt.
+   * Note that the attempt function will be invoked on the given execution context for subsequent
+   * tries and therefore must be thread safe (not touch unsafe mutable state).
+   *
+   * <b>Example usage:</b>
+   *
+   * //retry with back off
+   * {{{
+   * protected val sendAndReceive: HttpRequest => Future[HttpResponse]
+   * private val sendReceiveRetry: HttpRequest => Future[HttpResponse] = (req: HttpRequest) => retry[HttpResponse](
+   *   attempt = () => sendAndReceive(req),
+   *   attempts = 10,
+   *   delayFunction = attempted => Option(2.seconds * attempted)
+   *   scheduler = context.system.scheduler
```

Comment:
we can remove the scheduler here. It's implicit. Other code examples are not including the implicits args. 

If we keep it, we need to also pass `ExecutionContext` and on a separated arg block, ie: `(...)(context.system.dispatcher, context.system.scheduler)